### DISCUSSION TITLE
Add StrictNumber

### DISCUSSION THREAD

**Discussion by gespispace** [pydantic/pydantic#2750] [Question] [answered]
Hi, thanks for the cool library.

Now there is a set of StrictInt, StrictFloat and others.
But they only allow you to specify a number or a floating point number.
But do not allow a combination of these options to be specified.

Therefore, you cannot make the field possible 5 or 5.5.

We need StrictNumber, which takes 5 or 5.5 and nothing else.

**Comment by StephenBrown2** [comment, marked as answer]
Depending on your use case, that might be possible already with `Literal`, i.e. `Literal[5, 5.5]` (see https://pydantic-docs.helpmanual.io/usage/types/#literal-type) or a Union? i.e. `Union[StrictFloat, StrictInt]`

**Reply by gespispace** [reply]
Thanks. Union it is solve my problem.

**Reply by gespispace** [reply]
How to close my Feature Request?

**Reply by StephenBrown2** [reply]
You're welcome!

I converted this discussion to a question, and selected my comment as the answer. Should be good now!

**Comment by samuelcolvin** [comment]
Can you give some more context? In most cases `5` and `5.0` are equivalent in python.

Also, have you tried using decimal?
